### DISCUSSION TITLE
Lesson 12: Event Test case is failing

### DISCUSSION THREAD

**Discussion by talhaEth** [Cyfrin/foundry-full-course-cu#941] [Q&A] [answered]
```I am testing the event ` emit CollateralRedeemed(USER, USER, weth, AMOUNT_COLLATERAL);`
When I run `testEmitCollateralRedeemedWithCorrectsArgs` test case. it shows me this error.``` 
`[FAIL. Reason: Expected an emit, but the call reverted instead. Ensure you're testing the happy path when using the `expectEmit` cheatcode.]`
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/5945fb07-efe0-4f3f-ac5d-4fa84a02e25a)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/120d4a66-338f-4773-87d5-2057e5c650ea)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/11aead61-cb8b-49cf-99c4-7260c07d0911)

**Comment by theirrationalone** [comment, marked as answer]
According to log traces: it's indicating that you have different logs than the logs actually was expected. 

Look at here: 
```solidity
    event CollateralRedeemed(address indexed redeemFrom, address indexed redeemTo, address token, uint256 amount );  //if redeemFrom != redeemTo, then it wasw liquated

vm.expectEmit(true, true, true, true, address(dsce));
```

The third event data wasn't indexed which is `address token`.


We can do two things:

1. Let's make the `token` indexed at the definition of event (on both places DSCEngine.sol & DSCEngineTest.t.sol) then the logs will also have the expected third indexed log datum.

Or (simplest reso.) 

2. Let's not expect the third indexed value by making it false at all places where we emitted the event. (DSCEngine.sol & DSCEngine.t.sol Or somewhere else 😅) // hint for advanced testing. 😁

Hope You can resolve it now. Good Luck 🍀

Feel free to ask again if you have some doubts Or getting some crazy bugs... We all are here for US 😉

;`)

**Reply by talhaEth** [reply]
I am grateful for your assistance. Thank you very much.
I was using `address token` at dscEngine test file and `address indexed token` in my engine file.

**Reply by theirrationalone** [reply]
My pleasure Ser! ❤
